### PR TITLE
Mock validation

### PR DESCRIPTION
**Pull request by bergman**
# TL;DR
Add some simple validation to mocks
- [X] throw an exception if a mock was not used
- [X] throw an exception if a duplicate mock was already defined
- [ ] ~throw an exception if a mock was not found and not explicitly configured to run actual task~
- [X] test behaviour of unmocked SdkRunnableTask tasks (they run by default if task type has no mocks defined)

Improve error messages around mocks, 

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit-testing/src/main/java/org/flyte/flytekit/testing/SdkTestingExecutor.java [hunk 1]
```diff
@@ -368,17 +385,12 @@ public <InputT, OutputT> SdkTestingExecutor withWorkflowOutput(
...
-    TestingRunnableTask<InputT, OutputT> fixedTask =
-        getFixedTaskOrDefault(workflow.getName(), inputType, outputType);
-
     // replace workflow
     SdkWorkflow<InputT, OutputT> mockWorkflow =
         new TestingWorkflow<>(inputType, outputType, output);
 
     return toBuilder()
         .putWorkflowTemplate(workflow.getName(), mockWorkflow.toIdlTemplate())
-        .putFixedTask(workflow.getName(), fixedTask.withFixedOutput(input, output))
```
Comment (bergman) [inline flytekit-testing/src/main/java/org/flyte/flytekit/testing/SdkTestingExecutor.java:381]: What was this ever used for? Tests pass fine without this, and the additional validation I added in this PR failed with this line included.

**Comment by honnix** [comment]
I converted this to a draft, and please feel free to continue or close.

**Comment by RRap0so** [comment]
Closing since it's very out of date.  Feel free to re-open if still relevant.
